Shared/Samples/Apply hillshade renderer to raster/ApplyHillshadeRendererToRasterView.swift

```diff
@@ -0,0 +1,93 @@
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct ApplyHillshadeRendererToRasterView: View {
+    /// The model used to store the geo model and other expensive objects
+    /// used in this view.
+    class Model: ObservableObject {
+        /// The map that will be shown.
+        let map: Map
+        
+        /// The raster layer in the map.
+        private let rasterLayer: RasterLayer
+        
+        /// The raster renderer.
+        var renderer: HillshadeRenderer {
+            didSet {
+                // When the renderer is updated, update the layer accordingly.
+                rasterLayer.renderer = renderer
+            }
+        }
+        
+        init() {
+            // Gets the raster file URL.
+            let rasterFileURL = Bundle.main.url(forResource: "srtm", withExtension: "tiff", subdirectory: "srtm")!
+            
+            // Creates a raster with the file URL.
+            let raster = Raster(fileURL: rasterFileURL)
+            
+            // Creates a raster layer using the raster object.
+            rasterLayer = RasterLayer(raster: raster)
+            
+            // Apply the hillshade renderer to the raster layer.
+            renderer = HillshadeRenderer(
+                altitude: 45,
+                azimuth: 315,
+                slopeType: nil,
+                zFactor: 0.000016,
+                pixelSizeFactor: 1,
+                pixelSizePower: 1,
+                outputBitDepth: 8
+            )
+            rasterLayer.renderer = renderer
+            
+            // Create our map.
+            map = Map(basemap: .init(baseLayer: rasterLayer))
+        }
+    }
+    
+    /// The view model for the sample.
+    @StateObject private var model = Model()
```

Comment:
As Phil pointed out in https://github.com/Esri/arcgis-maps-sdk-swift-samples/pull/591#discussion_r2075950055 , the model can be simplified. Up to you.

Reply:
I went down that road for this sample as well but this sample also has to respond to renderer updates (from the settings view) which overcomplicated the code beyond what a model does. I feel this is the easiest code to understand from a sample perspective for this particular sample.

Reply:
It might actually be simpler than you think. For example in this sample, the renderer is created in two different places: once in the model and subsequent times in the settings view. See #589 for how that can be done with a single `onChange` modifier on the sample view itself, without using a view model.